libs/dyn/convert/normalize.go

```diff
@@ -93,12 +95,13 @@ func (n normalizeOptions) normalizeStruct(typ reflect.Type, src dyn.Value, seen
 					Severity: diag.Warning,
 					Summary:  fmt.Sprintf("unknown field: %s", pk.MustString()),
 					Location: pk.Location(),
+					Path:     path,
 				})
 				continue
 			}
 
 			// Normalize the value according to the field type.
-			nv, err := n.normalizeType(typ.FieldByIndex(index).Type, pv, seen)
+			nv, err := n.normalizeType(typ.FieldByIndex(index).Type, pv, seen, path.Append(dyn.Key(pk.MustString())))
```

Comment:
just to double check: Append does not mutate in place and returns a new copy, correct?

Reply:
Correct, as of #1295: https://github.com/databricks/cli/blob/8c144a2de4a160fc2bb1a0587f6177b68ee314dd/libs/dyn/path.go#L52-L59